authnz/src/main/java/io/stargate/auth/AuthorizationService.java

```diff
@@ -22,17 +22,69 @@
 
 public interface AuthorizationService {
 
+  /**
+   * Using the provided token will perform pre-authorization where possible, executes the query
+   * provided, and then authorizes the response of the query.
+   *
+   * @param action A {@link QueryBuilder} object to be executed and authorized against a token.
+   * @param token The authenticated token to use for authorization.
+   * @param primaryKeyValues A list of primary key values that will be used in the query and should
+   *     be authorized against the token.
+   * @param tableMetadata The {@link Table} that will be queried against.
+   * @return On success will return the result of the query and otherwise will return an exception
+   *     relating to the failure to authorize.
+   * @throws Exception An exception relating to the failure to authorize.
+   */
   ResultSet authorizedDataRead(
       Callable<ResultSet> action, String token, List<String> primaryKeyValues, Table tableMetadata)
```

Comment:
may be late at this point, but shouldn't `primaryKeyValues` be a `List<Object>`? Do we always encode all types as `String`?

Reply:
Realized I had to go a completely different direction with that in https://github.com/stargate/stargate/pull/386.

Reply:
:thinking: do we still want to merge this PR, then?